### PR TITLE
Dockerdockerdocker

### PR DESCRIPTION
**Pull request by rwaffen**
maybe we can use docker for integrations tests with travis and beaker

### CODE REVIEW

File: spec/docker-int/hiera/hiera.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+:backends:
+  - file
```
Comment (tuxmea) [inline spec/docker-int/hiera/hiera.yaml]: why file backend? it is not used.

File: spec/docker-int/hiera/hiera.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  - yaml
+:hierarchy:
+  - docker
+
+:yaml:
+    :datadir: '/etc/puppet/data'
+:file:
```
Comment (tuxmea) [inline spec/docker-int/hiera/hiera.yaml]: unused file backend

File: spec/docker-int/one-head-httpd.pp [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+
+class { 'one':
+  oned                => true,
+  node                => false,
+  sunstone            => true,
+  sunstone_passenger  => true,
+} ->
+
+class { 'httpd': }
```
Comment (tuxmea) [inline spec/docker-int/one-head-httpd.pp]: Usage of this file is missing in the examples in Readme

File: spec/docker-int/hiera/docker.yaml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+---
+one::head::ssh_pub_key: 'ssh pub key'
+one::head::ssh_priv_key: 'ssh-dsa priv key'
+one::oned::db: oneadmin
+one::oned::db_user: oneadmin
+one::oned::db_password: oneadmin
+one::oned::db_host: localhost
+one::oned::sunstone_listen_ip: '0.0.0.0'
```
Comment (tuxmea) [inline spec/docker-int/hiera/docker.yaml]: Is there a special reason why this is not part of docker directory?

File: README.md [hunk 5]
```diff
@@ -49,6 +49,32 @@ To deploy a Opennebula instance locally run:
...
+To deploy a Opennebula instance locally in a docker container run these commandos.
+
+Build base image with puppet in it (Depending on centos:latest):
+
+    docker build --rm -f docker/Dockerfile-Centos6 -t epost-dev/puppet .
+
+Build a image with the one puppet module in it:
+
+    docker build --rm -f docker/Dockerfile -t epost-dev/one .
+
```
Comment (tuxmea) [inline README.md]: Please provide a better description.
The first docker build command is needed to build an image with puppet installed.
The second command makes use of the image built in the first command.
This is not very obvious.

File: README.md [hunk 6]
```diff
@@ -49,6 +49,32 @@ To deploy a Opennebula instance locally run:
...
+Run puppet in the container, choose one:
+
+    docker run --rm epost-dev/one puppet apply /etc/puppet/modules/one/spec/docker-int/one-head.pp
+    docker run --rm epost-dev/one puppet apply /etc/puppet/modules/one/spec/docker-int/one-node.pp
+    docker run --rm epost-dev/one puppet apply /etc/puppet/modules/one/spec/docker-int/one-head-node.pp
+
+The files can be found in the spec directory of this project. One will build a one head,
+one will build a node and one a head which also can be a node.
+
```
Comment (tuxmea) [inline README.md]: Better description needed:
the three commands rely upon the second build command.
Add details of applied manifests by specifying which manifest will deliver which functionality.

**Comment by rwaffen** [comment]
Thanks for the feedback. Will fix that.

**Pull request merged**